src/main/kotlin/org/koitharu/kotatsu/parsers/site/wpcomics/vi/TopTruyen.kt

```diff
@@ -0,0 +1,283 @@
+				}
+			}
+			
+			if (filter.query?.isNotEmpty() == true) {
+				params.add("keyword=${filter.query.urlEncoded()}")
+			}
+			
+			if (page > 1) {
+				params.add("page=$page")
+			}
+			
+			if (params.isNotEmpty()) {
+				append("?")
+				append(params.joinToString("&"))
+			}
+		}
+		
+		val doc = webClient.httpGet(url).parseHtml()
+		return parseMangaList(doc)
+	}
+
+	private fun parseMangaList(doc: Document): List<Manga> {
+		return doc.select("div.item-manga").mapNotNull { div ->
+			val href = div.selectFirst("h3 a")?.attrAsRelativeUrl("href") ?: return@mapNotNull null
+			val tagElements = div.select("p.info a[href*=tim-truyen]")
+			val tags = tagElements.mapNotNullToSet { a ->
+				val key = a.attr("href").substringAfterLast("/")
+				availableTags().find { it.key == key }
+			}
+			
+			Manga(
+				id = generateUid(href),
+				url = href,
+				publicUrl = href.toAbsoluteUrl(div.host ?: domain),
+				coverUrl = div.selectFirst("div.image-item img")?.src().orEmpty(),
+				title = div.selectFirst("h3 a")?.text().orEmpty(),
+				altTitle = null,
+				rating = RATING_UNKNOWN,
+				tags = tags,
+				author = null,
+				state = null,
+				source = source,
+				isNsfw = isNsfwSource,
+			)
+		}
+	}
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val fullUrl = manga.url.toAbsoluteUrl(domain)
+		val doc = webClient.httpGet(fullUrl).parseHtml()
+		
+		val tagElements = doc.select("li.category.row p.detail-info a[href*=tim-truyen]")
+		val tags = tagElements.mapNotNullToSet { a ->
+			val key = a.attr("href").substringAfterLast("/")
+			availableTags().find { it.key == key }
+		}
+		
+		val description = doc.selectFirst("div.summary-content p.detail-summary")?.html().orEmpty()
+		val authorText = doc.selectFirst("li.author.row p.detail-info")?.text()?.trim().orEmpty()
+		
+		val author = if (authorText.isNotEmpty() && authorText != "Đang cập nhật") {
+			authorText
+		} else {
+			null
+		}
+		
+		val altTitleText = doc.selectFirst("li.name-other.row p.detail-info")?.text()?.trim()
+		val altTitle = if (!altTitleText.isNullOrEmpty() && altTitleText != "Đang cập nhật") {
+			altTitleText
+		} else {
+			null
+		}
+		
+		val stateText = doc.selectFirst("li.status.row p.detail-info span.label")?.text()?.trim()
+		val state = when (stateText) {
+			"Đang cập nhật" -> MangaState.ONGOING
+			"Đã hoàn thành" -> MangaState.FINISHED
+			else -> null
+		}
+		
+		return manga.copy(
+			author = author,
+			description = description,
+			state = state,
+			tags = tags,
+			chapters = getChapters(doc),
+			altTitle = altTitle
+		)
+	}
+
+	override suspend fun getChapters(doc: Document): List<MangaChapter> {
+		return doc.select("li.row:not([style*='display: none'])").mapChapters(reversed = true) { _, element ->
+			val chapterLink = element.selectFirst("a.chapter") ?: return@mapChapters null
+			val href = chapterLink.attr("href")?.toAbsoluteUrl(domain) ?: return@mapChapters null
+			val name = chapterLink.text().trim()
```

Comment:
```suggestion
			val name = chapterLink.text()
```
`text()` already trim string under-the-hood